swaybar/ipc.c

```diff
@@ -322,8 +335,11 @@ void ipc_bar_init(struct bar *bar, const char *bar_id) {
 	}
 	free(res);
 	json_object_put(outputs);
-
-	const char *subscribe_json = "[ \"workspace\", \"mode\" ]";
+	// Will need to be redone for display mode toggling compatibility
+	// Will require setting bar->config->hidden_state to "show" as default
+	const char *subscribe_json = strcmp(bar->config->display_mode, "hide") == 0 ?
+		"[ \"workspace\", \"mode\", \"modifier\" ]"
+		: "[ \"workspace\", \"mode\" ]";
```

Comment:
> Don't take that approach. This way the user can choose not to use this feature and adjust their security configuration accordingly.

I don't think security mandates one approach over another, the client must be free to ask for whatever it likes, It's the server that enforces policy and the client should deal. I think (and so did @calvinkosmatka, independently) that having the client ask for everything it wants, but use only what the server gives corresponds perfectly to the security model and would simplify matters.

This raises another point about client handling of security denials. In #1172 I saw that the server declined IPC, and swaybar carried on as if nothing happened, which I think is a problem.

That suggests that maybe the client should not subscribe to all events at once, but group them by their "severity". Some events are optional (modifier events), and some may be critical (abort if not available). 

For example, being assigned mode `hide` and then denied modifier events is an error event IMO.

Sorry for causing so much discussion and of course I don't mean calvin has to deal with all the secondary issues himself simply because his PR exposed them. I'm glad to do a followup for the things calvin wants to consider out of scope here, assuming @SirCmpwn is interested.

I'll switch to quiet mode now... :)